sktime/markov/msm/augmented_msm.py

```diff
@@ -0,0 +1,623 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__author__ = 'Simon Olsson'
+
+import logging
+from typing import Optional
+
+import numpy as np
+from msmtools import estimation as msmest
+from scipy.sparse import issparse
+
+from sktime.markov import TransitionCountModel
+from sktime.markov.msm import MarkovStateModel
+from sktime.util import confidence_interval, ensure_dtraj_list, ensure_traj_list
+from .._base import _MSMBaseEstimator
+
+
+class AMMOptimizerState(object):
+    r""" State of the optimization in AMMs. Is later attached to the MSM object for evaluation purposes. """
+
+    def __init__(self, expectations_by_state, experimental_measurements, experimental_measurement_weights,
+                 stationary_distribution, tensor_cache_size, symmetrized_count_matrix, count_matrix_row_sums):
+        r""" Creates a new AMM optimizer state.
+
+        Parameters
+        ----------
+        expectations_by_state : (N, K) ndarray
+            See :meth:`AugmentedMSMEstimator.__init__`.
+        experimental_measurements : (K,) ndarray
+            See :meth:`AugmentedMSMEstimator.__init__`.
+        experimental_measurement_weights : (K,) ndarray
+            See :meth:`AugmentedMSMEstimator.__init__`.
+        stationary_distribution : (N, N) ndarray
+            Reversibly estimated stationary distribution from count matrix.
+        tensor_cache_size : int
+            Number of slices of the R tensor simultaneously stored. This value is computed from the cache size
+            set in :meth:`AugmentedMSMEstimator.__init__`.
+        symmetrized_count_matrix : (N, N) ndarray
+            Symmetrized count matrix :math:`C_\mathrm{sym} = 0.5 (C + C^\top)` .
+        count_matrix_row_sums : (N, ) ndarray
+            Row sums of count matrix.
+        """
+        self.lagrange = np.zeros_like(experimental_measurements)
+        self.pi = stationary_distribution
+        self.pi_hat = np.copy(stationary_distribution)
+        self.m_hat = None
+        self.experimental_measurements = experimental_measurements
+        self.experimental_measurement_weights = experimental_measurement_weights
+        self.expectations_by_state = expectations_by_state
+        self.tensor_cache_size = tensor_cache_size
+        self.symmetrized_count_matrix = symmetrized_count_matrix
+        self.count_matrix_row_sums = count_matrix_row_sums
+        self.X = np.empty_like(symmetrized_count_matrix)
+
+        self.m_hat = np.empty(())
+        self._slope_obs = None
+        self.update_m_hat()
+        self.d_m_hat = 1e-1 * np.ones_like(self.m_hat)
```

Comment:
i'm good with the other variables but `d_m_hat` is tough for me

Reply:
Suggestions? The `d` in `d_m_hat` denotes a difference within the iteration. I thought this fairly standard.

Reply:
`delta` or `diff` instead of `d`, but also you can just keep it i don't mind

Reply:
delta it is